ipn/prefs.go

```diff
@@ -877,7 +878,7 @@ func (p *Prefs) ShouldWebClientBeRunning() bool {
 
 // PrefsFromBytes deserializes Prefs from a JSON blob.
 func PrefsFromBytes(b []byte) (*Prefs, error) {
-	p := NewPrefs()
```

Comment:
This looks wrong?

Reply:
Previously it would return the default prefs when b was empty.

Reply:
And I think it would apply the json as a mask

Reply:
Right so that confirms my fear that the JSON was indeed being applied as a patch so it does regress some implicit behavior unintentionally.

cc @bradfitz

Reply:
maybe you can change the signature to pass in `dst *Prefs` and then the callers can decide what they want?

Reply:
there are exactly two callers of this, one from `LoadPrefs` (which is a windows only utility, we could mark it as such) which could pass in `ipn.NewPrefs()` and the other is `loadSavedPrefs` which could pass in `p := ipn.NewPrefs(); p.NoStatefulFiltering = ""`

Reply:
ugh, I don't like it, but can't think of a better workaround.
`PrefsFromBytes` should really just be deserializing whatever is in those bytes without any extra defaulting. but changing it we risk some unsaved prefs fields that would be populated from `NewPrefs` defaults to reset to zero values.